### DISCUSSION TITLE
Activate clangd hints (code completion) on demand

### DISCUSSION THREAD

**Discussion by kocho1984** [clangd/clangd#2174] [Q&A]
Is there a way to show clangd hints only when a shortcut is used? I don't want to see those hints constantly, but rather on demand.

**Comment by HighCommander4** [comment]
This is a feature that's up to clients to provide.

VSCode provides it: you can enable it by setting `"editor.inlayHints.enabled": "offUnlessPressed"`; then the hints are only shown when you press Ctrl+Alt.

**Reply by kocho1984** [reply]
I'm referring to following hints. Option you provided doesn't affect them.
![image](https://github.com/user-attachments/assets/7fe9a121-4cbe-4e72-81da-478b9032b4c1)

It seems I called it wrongly. I'm referring to code completion.

**Reply by HighCommander4** [reply]
That is also controlled by the editor. For vscode the setting is `"editor.suggestOnTriggerCharacters": false`.

**Reply by kocho1984** [reply]
That stops showing code completion up after dot has been typed but following suggestions are still being showed during typing. I typed 'p' and I got 
![image](https://github.com/user-attachments/assets/1bc536dc-3391-4c3d-b1db-bb79239c817d)
